pandas/core/strings/accessor.py

```diff
@@ -2137,6 +2146,14 @@ def decode(self, encoding, errors: str = "strict"):
         2   ()
         dtype: object
         """
+        if (
+            dtype is not None
+            and not is_string_dtype(dtype)
+            and not is_object_dtype(dtype)
```

Comment:
FWIW, is_string_dtype currently (confusingly, sometimes, if you really only want StringDtype) also returns True for object dtype, so this final `and not is_object_dtype` is not strictly needed (although I find it useful for reading the code ;))